version: "3.9"
services:
  server:
    build:
      context: .
      dockerfile: ./server/Dockerfile
    env_file:
      - docker.env
    depends_on:
      - postgres
    ports:
      - "8080:8000"
    volumes:
      - /remote/xxx/films:/lewdlocale/films
      - /remote/xxx/errors:/lewdlocale/errors
      - /var/run/docker.sock:/var/run/docker.sock # to track other containers
    networks:
      lewdlocale:
    restart: on-failure:3
  
  indexer:
    build:
      context: .
      dockerfile: ./indexer/Dockerfile
    env_file:
      - docker.env
    depends_on:
      - postgres
    volumes:
      - /remote/xxx/errors:/lewdlocale/errors
    networks:
      lewdlocale:
    restart: on-failure:3
    
  downloader:
    build:
      context: .
      dockerfile: ./downloader/Dockerfile
    env_file:
      - docker.env
    depends_on:
      - postgres
    volumes:
      - /remote/xxx/films:/lewdlocale/films
      - /remote/xxx/errors:/lewdlocale/errors
    networks:
      lewdlocale:
    deploy:
      replicas: 5

  transcoder:
    build:
      context: .
      dockerfile: ./transcoder/Dockerfile
    env_file:
      - docker.env
    depends_on:
      - postgres
    volumes:
      - /remote/xxx/films:/lewdlocale/films
      - /remote/xxx/errors:/lewdlocale/errors
    networks:
      lewdlocale:
    restart: on-failure:3

    

  postgres:
    image: postgres:15.3-alpine
    env_file:
      - docker.env
    volumes:
      - /remote/xxx/database:/var/lib/postgresql/data # replace with on-system path
    ports:
      - "5433:5432" # for external access
    restart: on-failure:3
    hostname: postgres
    networks:
      lewdlocale:
    user: "3002:1000"
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U lewdlocale"]
      interval: 10s
      timeout: 5s
      retries: 5
  
networks:
  lewdlocale:
    driver: bridge